jsonrpc/filter_manager.go

```diff
@@ -296,37 +296,62 @@ func (f *FilterManager) dispatchEvent(evnt *blockchain.Event) error {
 		}
 	}
 
+	closedFilterIDs := make([]string, 0)
+
 	// flush all the websocket values
-	for _, filter := range f.filters {
+	for id, filter := range f.filters {
 		if filter.isWS() {
 			if flushErr := filter.flush(); flushErr != nil {
+				if errors.Is(flushErr, websocket.ErrCloseSent) {
+					closedFilterIDs = append(closedFilterIDs, id)
+
+					f.logger.Warn(fmt.Sprintf("Subscription %s has been closed", id))
+
+					continue
+				}
+
 				f.logger.Error(fmt.Sprintf("Unable to process flush, %v", flushErr))
 			}
 		}
 	}
 
+	f.filtersLock.RUnlock()
+
+	if len(closedFilterIDs) > 0 {
+		f.filtersLock.Lock()
+
+		for _, id := range closedFilterIDs {
+			f.removeFilterByID(id)
+		}
+
+		f.logger.Info(fmt.Sprintf("Removed %d filters due to closed connections", len(closedFilterIDs)))
+
+		f.filtersLock.Unlock()
+	}
+
 	return nil
 }
 
 func (f *FilterManager) Exists(id string) bool {
-	f.lock.Lock()
+	f.filtersLock.RLock()
 	_, ok := f.filters[id]
-	f.lock.Unlock()
+	f.filtersLock.RUnlock()
```

Comment:
Why not defer?

Reply:
Fixed for you :) https://github.com/0xPolygon/polygon-edge/commit/6f2fe32ceefcf128da94084513b34f02e1eacef0